homeassistant/components/shelly/__init__.py

```diff
@@ -409,6 +431,63 @@ def async_setup(self) -> None:
         self.device_id = entry.id
         self.device.subscribe_updates(self.async_set_updated_data)
 
+    async def async_trigger_ota_update(self, beta: bool = False) -> None:
+        """Trigger or schedule an ota update."""
+
+        async def _async_do_ota_update() -> None:
+            """Perform an ota update."""
+            beta_channel = self._ota_update_params[ATTR_BETA]
+            update_data = self.device.status["update"]
+            new_version = update_data["new_version"]
+            if beta_channel:
+                new_version = update_data["beta_version"]
+
+            _LOGGER.info(
+                "Start OTA update of device %s from '%s' to '%s'",
+                self.name,
+                update_data["old_version"],
+                new_version,
+            )
+            result = None
+            try:
+                async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
+                    result = await self.device.trigger_ota_update(beta=beta_channel)
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Error while perform ota update: %s", err)
+
+            _LOGGER.debug("Result of OTA update call: %s", result)
+
+        update_data = self.device.status["update"]
+        _LOGGER.debug("OTA update service - update_data: %s", update_data)
+
+        if self._ota_update_pending:
+            _LOGGER.error(
```

Comment:
I think this would be better logged as a warning

Reply:
sure, you are right - these are warnings, not errors!